### PR TITLE
remove annotation load-balancer-source-ranges

### LINKED ISSUE #50491: Remove old load-balancer-source-ranges annotation
This annotation has been converted to a field.

### PR DESCRIPTION
**Pull request by x1957**
**What this PR does / why we need it**:
revemo annotation load-balancer-source-ranges
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
fixes #50491
**Special notes for your reviewer**:
/assign @thockin
/sig network

**Release note**:

```release-note
Beta annotation `service.beta.kubernetes.io/load-balancer-source-ranges` have been removed. Please use fields `service.Spec.LoadBalancerSourceRanges` instead.
```

### CODE REVIEW

**Comment by xiangpengzhao** [comment]
/ok-to-test
please run `hack/update-bazel.sh` to update BUILD file(s).

**Comment by x1957** [comment]
@xiangpengzhao  thanks, nothing to update

**Comment by xiangpengzhao** [comment]
yeah that's because you don't modify package import after rebasing. When I first saw this PR, some package import was modified. In that case, that script should be run:)

**Comment by x1957** [comment]
@xiangpengzhao thanks^_^
@thockin Could you please review the code?

File: pkg/api/annotation_key_constants.go [hunk 1]
```diff
@@ -80,13 +80,4 @@ const (
...
 	LastAppliedConfigAnnotation = kubectlPrefix + "last-applied-configuration"
-
-	// AnnotationLoadBalancerSourceRangesKey is the key of the annotation on a service to set allowed ingress ranges on their LoadBalancers
-	//
-	// It should be a comma-separated list of CIDRs, e.g. `0.0.0.0/0` to
-	// allow full access (the default) or `18.0.0.0/8,56.0.0.0/8` to allow
-	// access only from the CIDRs currently allocated to MIT & the USPS.
-	//
-	// Not all cloud providers support this annotation, though AWS & GCE do.
-	AnnotationLoadBalancerSourceRangesKey = "service.beta.kubernetes.io/load-balancer-source-ranges"
```
Comment (xiangpengzhao) [inline pkg/api/annotation_key_constants.go]: please also remove `AnnotationLoadBalancerSourceRangesKey` in staging/src/k8s.io/api/core/v1/annotation_key_constants.go

Comment (x1957) [inline pkg/api/annotation_key_constants.go]: ok， I will do it later.

File: pkg/api/service/util.go [hunk 2]
```diff
@@ -53,16 +52,8 @@ func GetLoadBalancerSourceRanges(service *api.Service) (netsets.IPNet, error) {
...
-		if val == "" {
-			val = defaultLoadBalancerSourceRanges
-		}
-		specs := strings.Split(val, ",")
-		ipnets, err = netsets.ParseIPNets(specs...)
-		if err != nil {
-			return nil, fmt.Errorf("%s: %s is not valid. Expecting a comma-separated list of source IP ranges. For example, 10.0.0.0/24,192.168.2.0/24", api.AnnotationLoadBalancerSourceRangesKey, val)
-		}
+		specs := strings.Split(defaultLoadBalancerSourceRanges, ",")
+		ipnets, _ = netsets.ParseIPNets(specs...)
```
Comment (x1957) [inline pkg/api/service/util.go]: Should I return error here?
I think there is no error when parse defaultLoadBalancerSourceRanges("0.0.0.0/0").

File: pkg/api/service/util.go [hunk 3]
```diff
@@ -39,7 +38,6 @@ func IsAllowAll(ipnets netsets.IPNet) bool {
 }
 
 // GetLoadBalancerSourceRanges first try to parse and verify LoadBalancerSourceRanges field from a service.
```
Comment (xiangpengzhao) [inline pkg/api/service/util.go]: this comment should be modified.

File: pkg/api/v1/service/util.go [hunk 4]
```diff
@@ -39,7 +38,6 @@ func IsAllowAll(ipnets netsets.IPNet) bool {
 }
 
 // GetLoadBalancerSourceRanges first try to parse and verify LoadBalancerSourceRanges field from a service.
```
Comment (xiangpengzhao) [inline pkg/api/v1/service/util.go]: also this comment

File: pkg/api/validation/validation.go [hunk 5]
```diff
@@ -2947,17 +2947,13 @@ func ValidateService(service *api.Service) field.ErrorList {
...
 	}
 
-	// Validate SourceRange field and annotation
-	_, ok := service.Annotations[api.AnnotationLoadBalancerSourceRangesKey]
-	if len(service.Spec.LoadBalancerSourceRanges) > 0 || ok {
+	// Validate SourceRange field
+	if len(service.Spec.LoadBalancerSourceRanges) > 0 {
 		var fieldPath *field.Path
 		var val string
 		if len(service.Spec.LoadBalancerSourceRanges) > 0 {
```
Comment (xiangpengzhao) [inline pkg/api/validation/validation.go]: this condition is redundant

File: pkg/api/service/util.go (hunk 3 above)
Comment (xiangpengzhao) [inline pkg/api/service/util.go]: also remove the comment at L47. (not shown in review mode)

Comment (xiangpengzhao) [inline pkg/api/service/util.go]: i.e. `// if SourceRange field is specified, ignore sourceRange annotation`

File: pkg/api/v1/service/util.go (hunk 4 above)
Comment (xiangpengzhao) [inline pkg/api/v1/service/util.go]: and `// if SourceRange field is specified, ignore sourceRange annotation`

File: pkg/api/v1/service/util.go [hunk 6]
```diff
@@ -18,7 +18,6 @@ package service
 
 import (
 	"fmt"
-	"strings"
```
Comment (xiangpengzhao) [inline pkg/api/v1/service/util.go]: run `hack/update-bazel.sh` due to this deletion

**Review by xiangpengzhao — CHANGES_REQUESTED** [review]
some nits, otherwise LGTM. Tim please take another review.

**Comment by xiangpengzhao** [comment]
@thockin `/release-note-action-required` in comment isn't supported yet. Do we need `/release-note-action-required` for this PR? If so, please help label it. Thanks!

**Pull request closed without merging**